fog/ledger/server/src/key_image_service.rs

```diff
@@ -1,166 +1,107 @@
 // Copyright (c) 2018-2021 The MobileCoin Foundation
-
+use crate::server::DbPollSharedState;
 use fog_api::ledger_grpc::FogKeyImageApi;
-use fog_ledger_enclave::{
-    CheckKeyImagesResponse, KeyImageContext, KeyImageResult, KeyImageResultCode, LedgerEnclaveProxy,
-};
-use fog_ledger_enclave_api::Error as EnclaveError;
+use fog_ledger_enclave::LedgerEnclaveProxy;
+use fog_ledger_enclave_api::{Error as EnclaveError, UntrustedKeyImageQueryResponse};
 use grpcio::{RpcContext, RpcStatus, UnarySink};
-use mc_attest_api::attest::{AuthMessage, Message};
-use mc_attest_enclave_api::ClientSession;
+use mc_attest_api::{
+    attest,
+    attest::{AuthMessage, Message},
+};
 use mc_common::logger::{log, Logger};
 use mc_ledger_db::{self, Ledger};
 use mc_util_grpc::{
-    rpc_database_err, rpc_internal_error, rpc_invalid_arg_error, rpc_logger, rpc_permissions_error,
-    send_result, Authenticator,
+    rpc_internal_error, rpc_invalid_arg_error, rpc_logger, rpc_permissions_error, send_result,
+    Authenticator,
 };
 use mc_util_metrics::SVC_COUNTERS;
 use mc_watcher::watcher_db::WatcherDB;
-use mc_watcher_api::TimestampResultCode;
-use std::{convert::From, sync::Arc};
-
-// Maximum number of Key Images that may be checked in a single request.
-pub const MAX_REQUEST_SIZE: usize = 2000;
+use std::sync::{Arc, Mutex};
 
 #[derive(Clone)]
 pub struct KeyImageService<L: Ledger + Clone, E: LedgerEnclaveProxy> {
-    ledger: L,
-    watcher: WatcherDB,
+    pub ledger: L,
+    pub watcher: WatcherDB,
```

Comment:
do these need to be pub? it seems a bit funky

Reply:
The server.rs was cloning the key image service ledger and watcher when creating the db fetcher object

Reply:
I was able to make it not public